src/boilerplate/contract/solidity/nodes/ContractBoilerplateGenerator.ts

```diff
@@ -69,7 +69,7 @@ class ContractBoilerplateGenerator {
       } = scope;
 
       const fnDefBindings = scope.filterBindings(
-        (b: any) => b.kind === 'FunctionDefinition' && b.name !== '',
+        (b: any) => b.kind === 'FunctionDefinition' && b.name !== '' && b.path.containsSecret,
```

Comment:
When we have a completely public function it would get copied over but with proof, commitments etc. added to the fn signature - this fix makes sure those public functions don't get included in the enum used to index vks